packages-web/datagrid/src/Datagrid.tsx

```diff
@@ -46,6 +47,9 @@ export default function Datagrid(props: DatagridContainerProps): ReactElement {
             footerWidgets={<div className="header">{props.footerWidgets}</div>}
             hasMoreItems={props.datasource.hasMoreItems ?? false}
             headerWidgets={<div className="footer">{props.headerWidgets}</div>}
+            numberOfPages={
+                props.datasource.totalCount ? Math.ceil(props.datasource.totalCount / props.pageSize) : undefined
```

Comment:
You do not distinguish undefined from 0 here

Reply:
I am checking the number of pages in the pagination itself and if totalCount is undefined it will stay undefined.. otherwise calculate the pages

Reply:
If `props.datasource.totalCount` is `0` you will say that it is undefined, which is not correct.

Reply:
For pagination undefined = zero

Reply:
https://github.com/mendix/widgets-resources/blob/f1e5a24a2107cbf2e890b456dc410183105c8e0d/packages-web/datagrid/src/components/Pagination.tsx#L22

Reply:
This is not correct. Since it is pretty valid to have 0 pages - when you have no data. And it doesn't mean that you do not have paging enabled.

Reply:
Done